src/client/csfle/state_machine.rs

```diff
@@ -40,9 +40,8 @@ impl CryptExecutor {
         key_vault_namespace: Namespace,
         tls_options: Option<KmsProvidersTlsOptions>,
     ) -> Result<Self> {
-        let num_cpus = std::thread::available_parallelism()?.get();
```

Comment:
do we want to file a ticket / add a TODO about switching back to this once our MSRV is 1.59+?

Reply:
Good thought, done.